src/compiler/transformers/tsplus.ts

```diff
@@ -488,9 +501,69 @@ namespace ts {
                 return visitEachChild(node, visitor, context);
             }
         }
-        function getPathOfExtension(factory: NodeFactory, importer: TsPlusImporter, extension: { definition: SourceFile; exportName: string; }, source: SourceFile) {
+
+        function addUniqueLocalExtensionsVisitor(hoistedStatements: Array<Statement>, localUniqueExtensionNames: ESMap<string, Identifier>, context: TransformationContext) {
+            return function (node: Node): VisitResult<Node> {
+                switch (node.kind) {
+                    case SyntaxKind.FunctionDeclaration: {
+                        const declaration = node as FunctionDeclaration
+                        if (declaration.name) {
+                            const name = declaration.name.escapedText.toString()
+                            if (localUniqueExtensionNames.has(name)) {
+                                const uniqueIdentifier = localUniqueExtensionNames.get(name)!
+                                hoistedStatements.push(
+                                    context.factory.createVariableStatement(undefined, [
+                                        context.factory.createVariableDeclaration(
+                                            uniqueIdentifier,
+                                            undefined,
+                                            undefined,
+                                            context.factory.createIdentifier(name)
+                                        )
+                                    ])
+                                )
+                            }
+                        }
+                        return node;
+                    }
+                    case SyntaxKind.VariableStatement: {
+                        const variableStatement = node as VariableStatement;
+                        const declaration = variableStatement.declarationList.declarations[0];
+                        if (declaration && declaration.name && isIdentifier(declaration.name)) {
+                            const name = declaration.name.escapedText.toString();
+                            if (localUniqueExtensionNames.has(name)) {
+                                const uniqueIdentifier = localUniqueExtensionNames.get(name)!;
+                                hoistedStatements.push(
+                                    context.factory.createVariableStatement(undefined, [
+                                        context.factory.createVariableDeclaration(uniqueIdentifier)
+                                    ])
+                                );

```

Comment:
For VariableStatements, pushes only the declaration to `hoistedStatements`

Reply:
_but_, since these are _var_ declarations anyways, I guess it doesn't really matter if these are hoisted to the top 🙂